libraries/standard/http/src/http_client.c

```diff
@@ -220,25 +238,32 @@ static int findHeaderValueParserCallback( http_parser * pHttpParser,
  * the response. This callback is used to signal the parser to halt execution
  * if the requested header is not found.
  *
- * @param[in] pHttpParser The parser object to which this callback is registered.
+ * @param[in] pHttpParser The parsing context.
  *
- * @return Returns #HTTP_PARSER_STOP_PARSING for the parser to halt further execution,
- * as all headers have been parsed in the response.
+ * @return Returns #HTTP_PARSER_STOP_PARSING for the parser to halt further
+ * execution, as all headers have been parsed in the response.
  */
 static int findHeaderOnHeaderCompleteCallback( http_parser * pHttpParser );
 
 /*-----------------------------------------------------------*/
 
+/*-----------------------------------------------------------*/
+
 static uint8_t convertInt32ToAscii( int32_t value,
-                                    uint8_t * pBuffer,
+                                    char * pBuffer,
                                     size_t bufferLength )
 {
-    /* As input value may be altered and MISRA C 2012 rule 17.8 prevents modification
-     * of parameter, a local copy of the parameter is stored. */
-    uint32_t absoluteValue = 0u;
+    /* As input value may be altered and MISRA C 2012 rule 17.8 prevents
+     * modification of parameter, a local copy of the parameter is stored.
+     * absoluteValue stores the positive version of the input value. It's type
+     * remains the same type as the input value to avoid unnecessary casting on
+     * a privately used variable. This variables size will always be less
+     * than INT32_MAX. */
```

Comment:
Thanks for going in depth on this. I have a feeling this applies to some code in MQTT as well.

Reply:
I realized I put it's when it's suppose to be its lol